logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MessageSendingInterceptor.kt

```diff
@@ -9,18 +9,19 @@ import com.wire.kalium.logic.functional.map
 import com.wire.kalium.logic.util.MessageContentEncoder
 
 interface MessageSendingInterceptor {
-    suspend fun prepareMessage(message: Message.Regular): Either<CoreFailure, Message.Regular>
+    suspend fun prepareMessage(message: Message.Sendable): Either<CoreFailure, Message.Sendable>
 }
 
 class MessageSendingInterceptorImpl(
     private val messageContentEncoder: MessageContentEncoder,
     private val messageRepository: MessageRepository
 ) : MessageSendingInterceptor {
 
-    override suspend fun prepareMessage(message: Message.Regular): Either<CoreFailure, Message.Regular> {
+    override suspend fun prepareMessage(message: Message.Sendable): Either<CoreFailure, Message.Sendable> {
         val replyMessageContent = message.content
 
         if (replyMessageContent !is MessageContent.Text
+            || message !is Message.Regular
```

Comment:
Is that correct? I see we don't have Regular anymore!

Reply:
Regular still exists! Please look into Messages.kt for more details.


This Interceptor has the purpose of modifying text messages with quotes, calculating the hash of the quoted message, before sending.

If the message being sent isn't a text message (they must be regular with a text content), then just retur it without modifications.